src/Components/Books/Books.js

```diff
@@ -1,34 +1,39 @@
 import React, { Component } from 'react'
 import './Books.css'
-import { fetchAllBooks } from '../../API'
+import { fetchBooks } from '../../API'
 import Book from '../Book/Book'
-import ReadingList from '../ReadingList/ReadingList'
-import Search from '../Search/Search'
+
 import BookInfo from '../BookInfo/BookInfo'
 import { Route, Switch } from 'react-router-dom';
 import { connect } from 'react-redux'
-import { setBooks, addFavorite } from '../../actions';
+import { setBooks, addFavorite, setList } from '../../actions';
 import { bindActionCreators } from 'redux';
 
 class Books extends Component {
-  constructor() {
-    super()
+  constructor(props) {
+    super(props)
     this.state = {
+      books: [],
       foundBooks: [],
     }
   }
 
-  async componentDidMount() {
-    const { setBooks } = this.props;
-    try {
-      const books = await fetchAllBooks()
-      setBooks(books.results.books)
-    }
-    catch ({ message }) {
-      alert(message);
+  componentDidMount() {
+    this.setState({ books: this.props.filteredBooks })
+  }
```

Comment:
I think we may be able to get rid of this componentDidMount because the same thing is happening in componentDidUpdate now.